src/algorithms/pid_lut/PIDLookup.cc

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024, Nathan Brei, Dmitry Kalinkin
+
+#include <algorithms/service.h>
+#include <edm4eic/MCRecoParticleAssociationCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+#include <edm4hep/MCParticleCollection.h>
+#include <edm4hep/utils/vector_utils.h>
+#include <fmt/core.h>
+#include <podio/ObjectID.h>
+#include <cmath>
+#include <gsl/pointers>
+#include <stdexcept>
+
+#include "PIDLookup.h"
```

Comment:
Use full path under src/ to allow for easier future separation of headers from implementation.